articles/tutorials/advanced/2d_shaders/02_hot_reload/index.md

```diff
@@ -0,0 +1,280 @@
+
+[!code-xml[](./snippets/snippet-2-08.xml)]
+
+Now when you re-run the command from earlier, it will only run the `IncludeContent` target when `.fx` files have been changed. All edits to `.cs` files are ignored. Try adding a blank line to the `grayscaleEffect.fx` file, and notice the `dotnet watch` process re-build the content.
+
+However, if you would like to use `dotnet watch` for anything else in your workflow, then the configuration settings are too aggressive, because they will be applied _all_ invocations of `dotnet watch`. The `ItemGroup` can be optionally included when a certain condition is met. We will introduce a new MSBuild property called `OnlyWatchContentFiles`:
+
+[!code-xml[](./snippets/snippet-2-09.xml?highlight=1)]
+
+And now when `dotnet watch` is invoked, it needs to specify the new parameter:
+
+[!code-sh[](./snippets/snippet-2-10.sh)]
+
+The command is getting long and hard to type, and if we want to add more configuration, it will likely get even longer. Instead of invoking `dotnet watch` directly, it can be run as a new `<Target>` MSBuild step. Add this `<Target>` to your `DungeonSlime.csproj` file:
+
+[!code-xml[](./snippets/snippet-2-11.xml)]
+
+And now from the terminal, run the following `dotnet build` command:
+
+[!code-sh[](./snippets/snippet-2-12.sh)]
+
+
+We now have a way to dynamically recompile shaders on file changes and copy the `.xnb` files into the game folder! There are a few final adjustments to make to the configuration. 
+
+First, you may notice some odd characters in the log output after putting the `dotnet watch` inside the `WatchContent` target. This is because there are _emoji_ characters in the standard `dotnet watch` log stream, and some terminals do not understand how to display those, especially when streamed between `dotnet build`. To disable the _emoji_ characters, a `DOTNET_WATCH_SUPPRESS_EMOJIS` environment variable needs to be set:
+
+[!code-xml[](./snippets/snippet-2-13.xml?highlight=3)]
+
+Next, the `IncludeContent` target is doing a little too much work for our use case. It is trying to make sure the MonoGame Content Builder tools are installed. For our use case, we can opt out of that check by disabling the existing `AutoRestoreMGCBTool` MSBuild property. It also makes sense to pass `--restore:false` as well so that Nuget packages are not restored on each content file change:
+
+[!code-xml[](./snippets/snippet-2-14.xml?highlight=2)]
+
+To experiment with the system, re-run the following command:
+
+[!code-sh[](./snippets/snippet-2-15.sh)]
+
+And then cause some sort of compiler-error in the `grayscaleEffect.fx` file, such as adding the line, `"tunafish"` to the top of the file. When you save it, you should see the terminal spit out an error containing information about the compilation failure, 
```

Comment:
Actually, I think I have run into a problem here.

When I added the `WatchContent` block, I thought it was just a different build, but the process is continually building (counter going up)
Confirmed at this point because WatchContent is not finishing its build.

Ran `dotnet build` and the build finished (but errors thanks tot he shader change)

Can you re-test using ONLY the steps in this chapter from the original `27-conclusion` source.

Reply:
Just tried again with a fresh clone of the original tutorial, same result.

I also note that you GUM reference is NEWER than what is currently in the 2D tutorial, best not to change versions from the original (even if it has been updated) as the tutorial makes reference to specific versions.

Reply:
I'll discuss it with @vchelaru to see what should be used, as the 2D tutorial text and references needs to be updated with any version change

Reply:
Which versions are being discussed? The original 2D tutorial was written against version 5.31 I think but I’ve since updated the tutorial to a much newer version.